x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/ValuesDoubleAggregator.java

```diff
@@ -142,151 +263,85 @@ public void toIntermediate(Block[] blocks, int offset, IntVector selected, Drive
         }
 
         void addValue(int groupId, double v) {
-            values.add(groupId, Double.doubleToLongBits(v));
-            maxGroupId = Math.max(maxGroupId, groupId);
+            if (groupId > maxGroupId) {
+                firstValues = blockFactory.bigArrays().grow(firstValues, groupId + 1);
+                firstValues.set(groupId, v);
+                if (seen == null && groupId > maxGroupId + 1) {
```

Comment:
Maybe a comment that we don't build `seen` if the array is dense.

It should be super common that we're dense.